library/src/main/java/io/constructor/core/ConstructorIo.kt

```diff
@@ -160,6 +160,43 @@ object ConstructorIo {
         return dataManager.getAutocompleteResults(term.urlEncode(), encodedParams = encodedParams.toTypedArray())
     }
 
+    /**
+     * Returns a list of autocomplete suggestions
+     * ##Example
+     * ```
+     *  runBlocking {
+     *      launch {
+     *          try {
+     *              val autocompleteResults = constructorIo.getAutocompleteResultsCRT("Dav")
+     *              // Do something with autocompleteResults
+     *          } catch (e: Exception) {
+     *              println(e)
+     *          }
+     *      }
+     *  }
+     * ```
+     * @param term the term to search for
+     * @param facets additional facets used to refine results
+     * @param groupId category facet used to refine results
+     * @param hiddenFields show fields that are hidden by default
+     */
+    suspend fun getAutocompleteResultsCRT(term: String, facets: List<Pair<String, List<String>>>? = null, groupId: Int? = null, hiddenFields: List<String>? = null): AutocompleteResponse {
+        val encodedParams: ArrayList<Pair<String, String>> = arrayListOf()
+        groupId?.let { encodedParams.add(Constants.QueryConstants.FILTER_GROUP_ID.urlEncode() to it.toString()) }
+        facets?.forEach { facet ->
+            facet.second.forEach {
+                encodedParams.add(Constants.QueryConstants.FILTER_FACET.format(facet.first).urlEncode() to it.urlEncode())
+            }
+        }
+        configMemoryHolder.autocompleteResultCount?.entries?.forEach {
+            encodedParams.add(Pair(Constants.QueryConstants.NUM_RESULTS+it.key, it.value.toString()))
+        }
+        hiddenFields?.forEach { hiddenField ->
+            encodedParams.add(Constants.QueryConstants.FMT_OPTIONS.format(Constants.QueryConstants.HIDDEN_FIELD).urlEncode() to hiddenField.urlEncode())
+        }
```

Comment:
This part seems to be shared between CRT and non-CRT method (same in all the others methods named alike).
I'd prefer to see this extracted in a separate function that would be used on both places.
I don't see this as a blocker since we only have it twice, though.

Reply:
I'm going to hold off on this for now and create a story for us to look into extracting it. Thanks!